src/com/mellanox/cicd/Matrix.groovy

```diff
@@ -236,20 +240,45 @@ def runK8(image, branchName, config, axis) {
     }
 
     run_shell('printf "%s"' +  '"' + str + '"', "Matrix axis parameters")
-    
-
 
     def listV = parseListV(config.volumes)
     def cname = image.get("name").replaceAll("[\\.:/_]","")
-    def nodeSelector = getConfigVal(config, ['kubernetes', 'nodeSelector'], "")
-
-    podTemplate(cloud: cloudName, runAsUser: "0", runAsGroup: "0",
-                nodeSelector: nodeSelector,
-                containers: [
-                    containerTemplate(name: cname, image: image.url, ttyEnabled: true, alwaysPullImage: true, command: 'cat')
-                ],
-                volumes: listV
-                )
+
+    run_shell('printf "INFO: arch = %s"' + axis.arch, "DEBUG")
+
+    switch(axis.arch) {
+        case 'x86_64':
+            nodeSelector = 'kubernetes.io/arch=amd64'
```

Comment:
and here you can use 
```
nodeSelector = getConfigVal(config, ['kubernetes', 'arch_table', 'x86', 'nodeSelector']
```

* use function param instead of ```x86```

Reply:
Fixed.